framework/doc/content/documentation/framework_development/moose_utils.md

```diff
@@ -75,3 +75,124 @@ After the data is read using the "read" method, there are two methods used for a
+- `dsecondInvariant()`  derivatives of `secondInvariant()` wrt tensor entries
+- `d2secondInvariant()`  second derivatives of `secondInvariant()` wrt tensor entries
+- `thirdInvariant()` third invariant of the symmetric part of deviatoric, i.e. `((deviatoric() + deviatoric().transpose())/2).det()`
+- `dthirdInvariant()`  derivatives of `thirdInvariant()` wrt tensor entries
+- `d2thirdInvariant()`  second derivatives of `thirdInvariant()` wrt tensor entries
+- `sin3Lode()`  sine of three times the Lode angle
+- `dsin3Lode()`  derivatives of `sin3Lode()` wrt tensor entries
+- `d2sin3Lode()`  second derivatives of `sin3Lode()` wrt tensor entries
+- `det()` determinant
+- `ddet()` derivatives of `det()` wrt tensor entries
+- `inverse()`
+- `symmetricEigenvalues()`  eigenvalues of symmetric part of tensor
+- `dsymmetricEigenvalues()`  derivatives of symmetricEigenvalues wrt the tensor entries
+- `d2symmetricEigenvalues()` second derivatives of symmetricEigenvalues wrt the tensor entries
+
+These methods are thoroughly tested using CPPUNIT.
+
+### RankThreeTensor Operators
+
+The following operators are available for RankThreeTensor, with the values in parentheses indicating
+what the type of the other object to be subject to the operation.
+
+```text
+=
++= (RankThreeTensor)
+-= (RankThreeTensor)
+*= (Real)
+/= (Real)
++ (RankThreeTensor)
+- (RankThreeTensor)
+* (RankThreeTensor, RankTwoTensor, Real)
+/ (Real)
+```
+
+In addition, many methods are available for additional matrix operations:
+
+- `zero()`
+- `print()`
+- `L2norm()`
+- `rotate(Real)`
+- `rotate(RealTensorValue)`
+- `rotate(RankTwoTensor)`
+- fillFromInputVector(TypeVector<Real>)
+- `fillFromPlaneNormal(TypeVector<Real>)`
+- `mixedProjectRankFour(RankTwoTensor)` creates a Rank-4 tensor from the contraction $D_{ijkl} = T_{mij}b_{mn}T_{nkl}$
+- `doubleContraction(RankTwoTensor)` creates a vector from the contraction $v_i = T_{ijk}b_{jk}$
+
+### RankFourTensor Operators
+
+The following operators are available for RankFourTensor, with the values in parentheses indicating
+what the type of the other object to be subject to the operation.
+
+```text
+=
++= (RankFourTensor)
+-= (RankFourTensor)
+*= (Real)
+/= (Real)
++ (RankFourTensor)
+- (RankFourTensor)
+* (RankFourTensor, RankTwoTensor, RealTensorValue, Real)
+/ (Real)
+```
+
+In addition, many methods are available for additional matrix operations:
+
+- `zero()`
+- `print()`
+- `L2norm()`
+- `invSymm()` returns A$_{ijkl}$ from the input C$_{ijkl}$ such that $C_{ijkl} \cdot A_{ijkl} = 0.5 \left( \delta_{im} \delta_{jn} + \delta_{in} \delta_{jm} \right)$
+- `rotate(RealTensorValue)`
+- `rotate(RankTwoTensor)`
+- `transposeMajor()` witch first pair and second pair of indices
+- `surfaceFillFromInputVector(TypeVector<Real>)` Fills tensor entries by ignoring the last dimension
+- `fillFromInputVector(TypeVector<Real>, FillMethod)` used to build elasticity tensors, see [ComputeElasticityTensor](/Materials/ComputeElasticityTensor.md)
```

Comment:
@aeslaughter I'm at a loss for how to properly link this file. Would you tell me the magic combination for linking across different modules and/or the framework and a module please?